packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -190,6 +191,52 @@ export const getParticipants = async (req: Request, res: Response) => {
   }
 };
 
+export const updateParticipant = async (req: Request, res: Response) => {
+  try {
+    // valid giveaway
+    const giveaway = await Giveaway.findById(req.params.id);
+    if (!giveaway) return res.status(404).json({ error: 'Giveaway not found' });
+
+    // valid participant
+    const participant = giveaway.participants.find(
+      (participant) => participant.id === req.params.participantId
+    );
+    if (!participant)
+      return res.status(404).json({ error: 'Participant not found' });
+    if (participant.state !== ParticipantState.PENDING)
+      return res.status(400).json({ error: 'Participant state already set' });
+
+    // valid state
+    const { state } = req.body;
+    if (!Object.values(ParticipantState).includes(state))
+      return res
+        .status(400)
+        .json({ error: `${state} is not a valid participation state` });
```

Comment:
This validation is not needed here, the same thing is already being done by the Giveaway model and if an invalid state is provided an error will be thrown on save

Reply:
In that case we can expand the `handleError` function to also return a more appropriate status code for those cases. Something like this:
```javascript
// in utils/model.util.ts
import { Error as MongooseError } from 'mongoose';

type APIError = {
  code: number;
  message: object[] | string;
}

export const handleError = (error: Error): APIError  => {
  if (error instanceof MongooseError.ValidationError) {
    const message = Object.entries(error.errors).map(([field, error]) => ({
      field,
      message: error.message,
    }));
    return { code: 400, message }
  }
  return { code: 500, message: error.message };
};
```

And then in the controller, inside the catch condition:
```javascript
const { code, message } = handleError(error)
res.status(code).json({ error: message });
```

That way we can add more custom error handling while keeping our code DRY.